ext/crypto/00_crypto.js

```diff
@@ -2640,6 +2492,202 @@
+            break;
+          case 192:
+            jwk.alg = aesJwkAlg[algorithm.name][192];
+            break;
+          case 256:
+            jwk.alg = aesJwkAlg[algorithm.name][256];
+            break;
+          default:
+            throw new DOMException(
+              "Invalid key length",
+              "NotSupportedError",
+            );
+        }
+
+        // 5.
+        jwk.key_ops = key[_usages];
+        // 6.
+        jwk.ext = key[_extractable];
+        // 7.
+        return jwk;
+      }
+      default:
+        throw new DOMException("Not implemented", "NotSupportedError");
+    }
+  }
+
+  function importKeyAES(
+    format,
+    normalizedAlgorithm,
+    keyData,
+    extractable,
+    keyUsages,
+  ) {
+    // 2.
+    let data = keyData;
+    switch (format) {
+      case "raw": {
+        // 2.
+        if (
+          !ArrayPrototypeIncludes([128, 192, 256], keyData.byteLength * 8)
+        ) {
+          throw new DOMException("Invalid key length", "Datarror");
+        }
+
+        break;
+      }
+      case "jwk": {
+        // 1.
+        const jwk = keyData;
+        // 2.
+        if (jwk.kty !== "oct") {
+          throw new DOMException(
+            "`kty` member of JsonWebKey must be `oct`",
+            "DataError",
+          );
+        }
+
+        // Section 6.4.1 of RFC7518
+        if (jwk.k === undefined) {
+          throw new DOMException(
+            "`k` member of JsonWebKey must be present",
+            "DataError",
+          );
+        }
+
+        // 4.
+        data = decodeSymmetricKey(jwk.k);
+
+        // 5.
+        switch (data.byteLength * 8) {
+          case 128:
+            if (
+              jwk.alg !== undefined &&
+              jwk.alg !== aesJwkAlg[normalizedAlgorithm.name][128]
+            ) {
+              throw new DOMException("Invalid algorithm", "DataError");
+            }
+            break;
+          case 192:
+            if (
+              jwk.alg !== undefined &&
+              jwk.alg !== aesJwkAlg[normalizedAlgorithm.name][192]
+            ) {
+              throw new DOMException("Invalid algorithm", "DataError");
+            }
+            break;
+          case 256:
+            if (
+              jwk.alg !== undefined &&
+              jwk.alg !== aesJwkAlg[normalizedAlgorithm.name][256]
+            ) {
+              throw new DOMException("Invalid algorithm", "DataError");
+            }
+            break;
+          default:
+            throw new DOMException(
+              "Invalid key length",
+              "DataError",
+            );
+        }
+
+        // 6.
+        if (keyUsages.length > 0 && jwk.use && jwk.use !== "enc") {
+          throw new DOMException("Invalid key usages", "DataError");
+        }
+
+        // 7.
+        // Section 4.3 of RFC7517
+        if (jwk.key_ops) {
+          if (
+            ArrayPrototypeFind(
+              jwk.key_ops,
+              (u) => !ArrayPrototypeIncludes(recognisedUsages, u),
```

Comment:
This is quadratic although I expect key_ops and recognisedUsages are both small?

Same observation a few lines below.

Reply:
Yeah both are small, recognisedUsages.length is constant 8 and key_ops can be max 8